src/components/TeamMemberTasks/TeamMemberTask.jsx

```diff
@@ -84,10 +84,11 @@ const TeamMemberTask = React.memo(
     const isAllowedToResolveTasks =
       rolesAllowedToResolveTasks.includes(userRole) || dispatch(hasPermission('resolveTask'));
     const isAllowedToSeeDeadlineCount = rolesAllowedToSeeDeadlineCount.includes(userRole);
+    const isAllowedToEditTask = rolesAllowedToResolveTasks.includes(userRole);
     // ^^^
 
     const canGetWeeklySummaries = dispatch(hasPermission('getWeeklySummaries'));
-    const canUpdateTask = dispatch(hasPermission('updateTask'));
+    const canUpdateTask = dispatch(hasPermission('updateTask')) || isAllowedToEditTask;
```

Comment:
```suggestion
    const canUpdateTask = dispatch(hasPermission('updateTask'))
```
1. Creating hard-coded role checks like this has caused a lot of confusion in the past. It should only be handled through permissions.
2. `isAllowedToEditTask` is for Admins and Owners. Both these roles should be guaranteed the `updateTask` permission, so this logic is redundant.
3. This doesn't seem to address the Mentor concerns in the listed bug report.

Reply:
Shouldn't this bug be able to be fixed in-app through granting Managers the permission in Permissions Management?